internal/server/statvar_group.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/datacommonsorg/mixer/internal/util"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func filterSVG(svgResp *pb.StatVarGroups, placeSVs []string) *pb.StatVarGroups {
+	// Build set for all the SV.
+	svSet := map[string]struct{}{}
+	for _, sv := range placeSVs {
+		svSet[sv] = struct{}{}
+	}
+
+	// All the svg with valid sv for this place
+	svgSet := map[string]struct{}{}
+
+	// Iterate over stat var group, and only keep children with valid stat vars
+	// for this place.
+	for svgID, svgData := range svgResp.StatVarGroups {
+		filteredChildren := []string{}
+		for _, child := range svgData.ChildStatVars {
+			if _, ok := svSet[child]; ok {
+				filteredChildren = append(filteredChildren, child)
+			}
+		}
+		svgData.ChildStatVars = filteredChildren
+		if len(filteredChildren) > 0 {
```

Comment:
There could be grandchildren and below without children? so should we check all child SVGs are excluded before excluding?

Reply:
I see. Its handled in the next loop.